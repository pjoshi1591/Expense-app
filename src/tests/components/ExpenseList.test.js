import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('Should render ExpenseList with expense', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('SHould render ExpenseList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});