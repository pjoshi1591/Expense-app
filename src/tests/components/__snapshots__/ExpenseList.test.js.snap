// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHould render ExpenseList with empty message 1`] = `
<div>
  <p>
    No expense
  </p>
</div>
`;

exports[`Should render ExpenseList with expense 1`] = `
<div>
  <ExpenseListItem
    amount={400}
    createdAt={0}
    description="Rent"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={50}
    createdAt={-345600000}
    description="Water"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={20}
    createdAt={345600000}
    description="Internet"
    id="3"
    key="3"
    note=""
  />
</div>
`;
